crates/matrix-sdk-crypto/src/verification/requests.rs

```diff
@@ -634,6 +647,20 @@ impl VerificationRequest {
                 if let Some((sas, content)) =
                     s.clone().start_sas(self.we_started, self.inner.clone().into()).await?
                 {
+                    // We may have previously started QR verification and generated a QR code. If we
+                    // now switch to SAS flow, the previous verification has to be replaced
+                    #[cfg(feature = "qrcode")]
+                    if let Some(_) = self
+                        .verification_cache
+                        .get_qr(sas.other_user_id().clone(), sas.flow_id().as_str())
+                    {
+                        info!("We have an ongoing QR verification, replacing with SAS");
+                        self.verification_cache.replace(sas.clone().into())
+                    } else {
+                        self.verification_cache.insert_sas(sas.clone());
+                    }
+
+                    #[cfg(not(feature = "qrcode"))]
```

Comment:
Not sure if there is a better syntax for the else branch. I tried `if cfg!(feature = "qrcode") {} else {}` but that was still complaining about unavailable `get_qr` method

Reply:
`if cfg!()` is used when you have the feature enabled but the codepath differs, all code paths still need to compile. 

It's commonly used to have different codepaths based on the architecture or the OS, see for example this: https://doc.rust-lang.org/std/macro.cfg.html#examples.

Reply:
So is the current branching the most efficient?
```rust
if cfg!(feature = "qrcode") {
    #[cfg(feature = "qrcode")]
    if self
        .verification_cache
        .get_qr(sas.other_user_id(), sas.flow_id().as_str())
        .is_some()
    {
        ...
    } else {
        ...
    }
} else {
    ...
}
```

Reply:
> So is the current branching the most efficient?
> 
> ```rust
> if cfg!(feature = "qrcode") {
>     #[cfg(feature = "qrcode")]
>     if self
>         .verification_cache
>         .get_qr(sas.other_user_id(), sas.flow_id().as_str())
>         .is_some()
>     {
>         ...
>     } else {
>         ...
>     }
> } else {
>     ...
> }
> ```

As mentioned, the `cfg!()` doesn't seem to be needed, this is what I had in mind:

```rust
// We may have previously started QR verification and generated a QR code. If we
// now switch to SAS flow, the previous verification has to be replaced
#[cfg(feature = "qrcode")]
if self
    .verification_cache
    .get_qr(sas.other_user_id(), sas.flow_id().as_str())
    .is_some()
{
    info!("We have an ongoing QR verification, replacing with SAS");
    self.verification_cache.replace(sas.clone().into())
} else {
    self.verification_cache.insert_sas(sas.clone());
}

#[cfg(not(feature = "qrcode"))]
self.verification_cache.insert_sas(sas.clone());
```

Reply:
Yea this is exactly what I wrote previously, I am just concerned about the gap between the if/else branch and the `not(feature)` ... one could easily add more code in between and then the logical relation between the feature / not-feature would be less apparent. Using `if cfg!(feature = "qrcode") {` is longer but gives very explicit `else` branch